modules/apps/object/object-service/src/main/java/com/liferay/object/internal/action/executor/NotificationTemplateObjectActionExecutorImpl.java

```diff
@@ -63,32 +59,53 @@ public void execute(
 				_dtoConverterRegistry, objectDefinition, payloadJSONObject,
 				_systemObjectDefinitionManagerRegistry));
 
-		notificationType.sendNotification(
-			new NotificationContextBuilder(
-			).className(
-				objectDefinition.getClassName()
-			).classPK(
-				GetterUtil.getLong(termValues.get("id"))
-			).externalReferenceCode(
-				GetterUtil.getString(termValues.get("externalReferenceCode"))
-			).groupId(
-				GetterUtil.getLong(termValues.get("groupId"))
-			).notificationTemplate(
-				notificationTemplate
-			).termValues(
-				termValues
-			).userId(
-				userId
-			).portletId(
-				objectDefinition.isUnmodifiableSystemObject() ?
-					StringPool.BLANK : objectDefinition.getPortletId()
-			).preferredLanguageId(
-				payloadJSONObject.getString("preferredLanguageId")
-			).usePreferredLanguageForGuests(
-				GetterUtil.getBoolean(
-					parametersUnicodeProperties.get(
-						"usePreferredLanguageForGuests"))
-			).build());
+		UserNotificationDelivery userNotificationDelivery =
+			_userNotificationDeliveryLocalService.getUserNotificationDelivery(
+				userId, objectDefinition.getPortletId(),
+				_classNameLocalService.getClassNameId(
+					objectDefinition.getClassName()),
+				UserNotificationDefinition.NOTIFICATION_TYPE_UPDATE_ENTRY,
+				UserNotificationDeliveryConstants.TYPE_WEBSITE, true);
+
+		if (userNotificationDelivery.isDeliver()) {
```

Comment:
@mariuo also we have to provide tests to avoid that this situation occur in the future, I recommend you create integration tests, since you are modifying the `UserNotificationType` class you can create the test in [`UserNotificationTypeTest`](https://github.com/liferay/liferay-portal/blob/65e821b7fc45365d39ca7c39c2fae49309b0f20b/modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/UserNotificationTypeTest.java#L74), you can create a test with the following name, `testIsDelivery`, take a look at other tests of this class, if you have questions please reach me o Slack.

Thank you!